File: client/src/store/State.ts
import { Game } from './Game';
import { Hand, Result } from './Hand';
import { Player } from './Player';

export interface State {
  readonly currentGame: Game;
  readonly playersHands: Array<Hand>;
  readonly dealersHand: Hand;
  readonly activeHand: number;
  readonly activeUsers: Array<Player>;
  readonly turnIsGoing: boolean;
}

export const defaultState = {
  currentGame: { players: [] } as Game,
  playersHands: [],
  dealersHand: {
    cards: [],
    player: 'dealer',
    bet: 0,
    isStaying: false,
    result: Result.PLAYING,
  } as Hand,
  activeHand: 0,
  activeUsers: [],
  turnIsGoing: false,
} as State;

File: rails/app/data/state.rb
# frozen_string_literal: true

class State

  attr_accessor :current_game, :players_hands, :dealers_hand, :active_hand, :active_users, :turn_is_going

  @current_game
  @players_hands
  @dealers_hand
  @active_hand
  @active_users
  @turn_is_going

  def initialize
    super
    @current_game = Game.new
    @players_hands = []
    @dealers_hand = Hand.new("dealer")
    @active_hand = 0
    @active_users = []
    @turn_is_going = false
  end
end
